source/lac/petsc_parallel_sparse_matrix.cc

```diff
@@ -406,6 +428,218 @@ namespace PETScWrappers
         }
     }
 
+    // BDDC
+    template <typename SparsityPatternType>
+    void
+    SparseMatrix::do_reinit_IS(const IndexSet &           local_rows,
+                               const IndexSet &           local_active_rows,
+                               const IndexSet &           local_columns,
+                               const IndexSet &           local_active_columns,
+                               const SparsityPatternType &sparsity_pattern)
+    {
+      Assert(sparsity_pattern.n_rows() == local_rows.size(),
+             ExcMessage(
+               "SparsityPattern and IndexSet have different number of rows."));
+      Assert(
+        sparsity_pattern.n_cols() == local_columns.size(),
+        ExcMessage(
+          "SparsityPattern and IndexSet have different number of columns"));
+      Assert(local_rows.is_contiguous() && local_columns.is_contiguous(),
+             ExcMessage("PETSc only supports contiguous row/column ranges"));
+      Assert(local_rows.is_ascending_and_one_to_one(communicator),
+             ExcNotImplemented());
+
+#  ifdef DEBUG
+      {
+        // check indexsets
+        types::global_dof_index row_owners =
+          Utilities::MPI::sum(local_rows.n_elements(), communicator);
+        types::global_dof_index col_owners =
+          Utilities::MPI::sum(local_columns.n_elements(), communicator);
+        Assert(row_owners == sparsity_pattern.n_rows(),
+               ExcMessage(
+                 std::string(
+                   "Each row has to be owned by exactly one owner (n_rows()=") +
+                 std::to_string(sparsity_pattern.n_rows()) +
+                 " but sum(local_rows.n_elements())=" +
+                 std::to_string(row_owners) + ")"));
+        Assert(
+          col_owners == sparsity_pattern.n_cols(),
+          ExcMessage(
+            std::string(
+              "Each column has to be owned by exactly one owner (n_cols()=") +
+            std::to_string(sparsity_pattern.n_cols()) +
+            " but sum(local_columns.n_elements())=" +
+            std::to_string(col_owners) + ")"));
+      }
+#  endif
+      PetscErrorCode ierr;
+
+      // create the local to global mappings as arrays.
+      IndexSet::size_type n_local_active_rows = local_active_rows.n_elements();
+      IndexSet::size_type n_local_active_cols =
+        local_active_columns.n_elements();
+      std::vector<int> idx_glob_row(n_local_active_rows);
+      std::vector<int> idx_glob_col(n_local_active_cols);
```

Comment:
```suggestion
      std::vector<PetscInt> idx_glob_row(n_local_active_rows);
      std::vector<PetscInt> idx_glob_col(n_local_active_cols);
```
(since these are passed to a PETSc function)

Reply:
Changed it as suggested